### PR TITLE
fix 404 firefox

### PR DESCRIPTION
**Pull request by easteregg**

### CODE REVIEW

File: src/root_files/_common/404.html [hunk 1]
```diff
@@ -97,11 +97,12 @@
             }
         };
     }
-
     var all_lang_regex = new RegExp('(' + all_languages.join('|') + ')', 'i');
     var is_from_404 = location.search &&
         location.search.toLowerCase().indexOf('from_404')  !== -1;
-    var redirectFn = routeHandler.isApp2() && !is_from_404 ? routeHandler.redirectToReactRoute : routeHandler.redirect;
+    var is_from_app2 = location.search &&
+        location.search.toLowerCase().indexOf('app2')  !== -1;
```
Comment (negarn) [inline src/root_files/_common/404.html]: extra space before `!` for this var and above var

File: src/root_files/_common/404.html [hunk 2]
```diff
@@ -97,11 +97,12 @@
...
+    var redirectFn = routeHandler.isApp2() && !is_from_404 ? routeHandler.redirectToReactRoute : is_from_app2 ? routeHandler.redirectToReactRoute : routeHandler.redirect;
```
Comment (negarn) [inline src/root_files/_common/404.html]: please use brackets to indicate which statements go together

Comment (negarn) [reply src/root_files/_common/404.html]: actually, can't this be changed to:
```
var redirectFn = ((routeHandler.isApp2() && !is_from_404) || is_from_app2) ? routeHandler.redirectToReactRoute : routeHandler.redirect;
```

Comment (easteregg) [reply src/root_files/_common/404.html]: Absolutely. Done.

**Pull request closed without merging**